### PR TITLE
Improve type of some utils function

### PR DESCRIPTION
**Pull request by nd-02110114**
I improve the type annotation and file name in `deepchem.utils` except `conformers.py`, `evaluate.py`, `rdkit_util.py` and `save.py`

I will make an another PR about `conformers.py`, `evaluate.py`, `rdkit_util.py` and `save.py`.

### CODE REVIEW

**Comment by nd-02110114** [comment]
This PR is ready to review!

File: docs/utils.rst [hunk 1]
```diff
@@ -137,11 +137,9 @@ Evaluation Utils
 Genomic Utilities
 -----------------
 
-.. autofunction:: deepchem.utils.genomics.seq_one_hot_encode
+.. autofunction:: deepchem.utils.genomics_utils.seq_one_hot_encode
 
-.. autofunction:: deepchem.utils.genomics.encode_fasta_sequence
```
Comment (rbharath) [inline docs/utils.rst:142]: Could you add in
```
.. autofunction:: deepchem.utils.genomic_utils.encode_fasta_sequence
```

File: deepchem/utils/mol_xyz_util.py [hunk 2]
```diff
@@ -1,13 +0,0 @@
-import numpy as np
```
Comment (nd-02110114) [inline deepchem/utils/mol_xyz_util.py:1]: This filae is out-of-date. I removed.

File: docs/utils.rst (hunk 1 above)
Comment (nd-02110114) [reply docs/utils.rst:142]: I think the `encode_fasta_sequence ` is alomost same API as `encode_bio_sequence`. So, I removed it. For the backward compatibility, I export `encode_fasta_sequence` from `deepchem/utils/save.py`

If you think it is better to export  `encode_fasta_sequence `, I will fix this point 🙆‍♂️

Comment (rbharath) [reply docs/utils.rst:142]: Ah that makes sense! Should be all good then :)

**Pull request merged**